src/main/java/edu/wisc/my/personalizedredirection/service/RedirectionServiceImpl.java

```diff
@@ -64,8 +68,8 @@ private IRedirectURLSourceDataParser getParser(UrlDataSource dataSource) throws
 		}
 
 		if (parser == null) {
-		    logger.error("Attempted to instantiate a DataSource with type " + dataSource.getDataSourceType());
-			throw new RuntimeException("Invalid data source type");
```

Comment:
If we're no longer throwing this error, I think we can remove the `throws RuntimeException`  from the signature